File: models/quakes.go
package models

import "time"

type Quake struct {
	ID             string    `json:"id"`
	Place          string    `json:"place"`
	Magnitude      float64   `json:"magnitude"`
	Depth          float64   `json:"depth"`
	Lat            float64   `json:"lat"`
	Lon            float64   `json:"lon"`
	Time           time.Time `json:"time"`
	Alert          string    `json:"alert"`
	Tsunami        int       `json:"tsunami"`
	MagnitudeLevel string    `json:"magnitude_level"`
	URL            string    `json:"url"`
}

type QuakesResponse struct {
	Total       int       `json:"total"`
	Quakes      []Quake   `json:"quakes"`
	LastUpdated time.Time `json:"last_updated"`
}

func GetMagnitudeLevel(mag float64) string {
	if mag >= 6.0 {
		return "major"
	} else if mag >= 5.0 {
		return "strong"
	} else if mag >= 4.0 {
		return "moderate"
	} else if mag >= 3.0 {
		return "minor"
	}
	return "unknown"
}

File: frontend/stores/quakes.ts
import { defineStore } from 'pinia'

export interface Quake {
  id: string
  place: string
  magnitude: number
  depth: number
  lat: number
  lon: number
  time: string
  magnitudeLevel: string
  tsunami: number
  url: string
}

export interface QuakesResponse {
  total: number
  quakes: Quake[]
  lastUpdated: string
}

export const useQuakesStore = defineStore('quakes', {
  state: () => ({
    data: null as QuakesResponse | null,
    loading: false,
    error: null as string | null
  }),

  actions: {
    async fetchQuakes() {
      this.loading = true
      this.error = null
      try {
        const config = useRuntimeConfig()
        const apiBase = config.public.apiBase

        interface RawQuake {
          id: string
          place: string
          magnitude: number
          depth: number
          lat: number
          lon: number
          time: string
          alert: string
          tsunami: number
          magnitude_level: string
          url: string
        }

        interface RawQuakesResponse {
          total: number
          quakes: RawQuake[]
          last_updated: string
        }

        const raw = await $fetch<RawQuakesResponse>(`${apiBase}/api/v1/quakes`)

        this.data = {
          total: raw.total,
          quakes: raw.quakes.map(q => ({
            id: q.id,
            place: q.place,
            magnitude: q.magnitude,
            depth: q.depth,
            lat: q.lat,
            lon: q.lon,
            time: q.time,
            magnitudeLevel: q.magnitude_level,
            tsunami: q.tsunami,
            url: q.url
          })),
          lastUpdated: raw.last_updated
        }
      } catch (err: any) {
        console.error('Error fetching quakes:', err)
        this.error = err.message || 'Failed to fetch quakes'
      } finally {
        this.loading = false
      }
    }
  }
})
